core/src/main/java/com/sequenceiq/cloudbreak/repository/ClusterDtoRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.sequenceiq.cloudbreak.repository;
+
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.Repository;
+import org.springframework.data.repository.query.Param;
+
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.view.ClusterView;
+import com.sequenceiq.cloudbreak.view.delegate.ClusterViewDelegate;
+
+@org.springframework.stereotype.Repository
+public interface ClusterDtoRepository extends Repository<Cluster, Long> {
+
+    @Query("SELECT c.id as id, " +
+            "c.creationFinished as creationFinished, " +
+            "c.creationStarted as creationStarted, " +
+            "c.upSince as upSince, " +
+            "c.fqdn as fqdn, " +
+            "c.clusterManagerIp as clusterManagerIp, " +
+            "c.name as name, " +
+            "c.description as description, " +
+            "c.databaseServerCrn as databaseServerCrn, " +
+            "c.rangerRazEnabled as rangerRazEnabled, " +
+            "c.certExpirationState as certExpirationState, " +
+            "c.environmentCrn as environmentCrn, " +
+            "c.proxyConfigCrn as proxyConfigCrn, " +
+            "c.uptime as uptime, " +
+            "c.extendedBlueprintText as extendedBlueprintTextSecret, " +
+            "c.attributes as attributesSecret, " +
+            "c.customContainerDefinition as customContainerDefinition, " +
+            "c.dpAmbariUser as dpAmbariUserSecret, " +
+            "c.dpAmbariPassword as dpAmbariPasswordSecret, " +
+            "c.password as passwordSecret, " +
+            "c.userName as userNameSecret, " +
+            "c.executorType as executorType, " +
+            "c.variant as variant, " +
+            "c.cloudbreakAmbariUser as cloudbreakAmbariUserSecret, " +
+            "c.cloudbreakAmbariPassword as cloudbreakAmbariPasswordSecret, " +
+            "c.cdpNodeStatusMonitorPassword as cdpNodeStatusMonitorPasswordSecret, " +
+            "c.cdpNodeStatusMonitorUser as cdpNodeStatusMonitorUserSecret, " +
+            "c.cloudbreakClusterManagerMonitoringPassword as cloudbreakClusterManagerMonitoringPasswordSecret, " +
+            "c.cloudbreakClusterManagerMonitoringUser as cloudbreakClusterManagerMonitoringUserSecret, " +
+            "c.databusCredential as databusCredentialSecret, " +
+            "c.embeddedDatabaseOnAttachedDisk as embeddedDatabaseOnAttachedDisk, " +
+
+            "c.keyStorePwd as keyStorePwdSecret, " +
+            "c.trustStorePwd as trustStorePwdSecret, " +
+
+            "f as fileSystem, " +
+            "af as additionalFileSystem, " +
+            "c.autoTlsEnabled as autoTlsEnabled " +
+            "FROM Cluster c " +
+            "LEFT JOIN c.stack s " +
```

Comment:
I think most of cases the left join enough. We want the cluster if exist by stack id. it would be affected if we list the records from the table but we add the where clause. 

However, I will try it without the stack join, If it works I will remove it

Reply:
I remember, the join is mandatory to filter by the stack id in JPQL. However it is unnecessary in native query